kolibri/core/assets/src/core-app/index.js

```diff
@@ -71,6 +72,23 @@ heartbeat.startPolling();
 
 i18nSetup().then(coreApp.ready);
 
+// these shall be responsibe for catching runtime errors
+Vue.config.errorHandler = function(err) {
+  logging.error(`Unexpected Error: ${err}`);
+  ErrorReportResource.report(err);
+};
+
+window.addEventListener('error', e => {
+  logging.error(`Unexpected Error: ${e.error}`);
+  ErrorReportResource.report(e.error);
```

Comment:
I think for purposes of context it would be good to get the entire error object?

Reply:
It would be a good thing. But the report() will have too many conditions to make the code look not so clean. Eg:

1. The Vue handler object will be handled with e.message and e.stack
2. The error event will be handled with e.error.message and e.error.stack
3. The promise rejection will be handled as e.reason.message and e.reason.stack

Reply:
In the later stage, we can do that. We will be optimizing the model anyway.

Reply:
Alright I will do the changes